scripts/globals/spells/sneak.lua

```diff
@@ -1,33 +1,29 @@
 -----------------------------------------
 -- Spell: Sneak
--- Lessens chance of being detected by sound
--- Duration is random number between 30 seconds and 5 minutes
+-- Lessens chance of being detected by sound.
+-- Duration is random number between 30 seconds and 5 minutes.
 -----------------------------------------
 require("scripts/globals/magic")
 require("scripts/globals/msg")
 require("scripts/globals/settings")
 require("scripts/globals/status")
+-----------------------------------------
 
-function onMagicCastingCheck(caster,target,spell)
+function onMagicCastingCheck(caster, target, spell)
     return 0
 end
 
-function onSpellCast(caster,target,spell)
-    if (not target:hasStatusEffect(dsp.effect.SNEAK)) then
+function onSpellCast(caster, target, spell)
+    if not target:hasStatusEffect(dsp.effect.SNEAK) then
 
-        -- last 7-9 minutes
-        local duration = calculateDuration(math.random(420, 540), spell:getSkillType(), spell:getSpellGroup(), caster, target)
-        duration = calculateDurationForLvl(duration, 20, target:getMainLvl())
+        local duration = calculateDuration(math.random(42, 54) * 10, spell:getSkillType(), spell:getSpellGroup(), caster, target)
 
-        -- FIXME: Create mod and use that instead of an itemID check
-        if (target:getEquipID(dsp.slot.BACK) == 13692) then -- skulker's cape
-            duration = duration * 1.5
-        end
+        duration = calculateDurationForLvl(duration, 20, target:getMainLvl()) + player:getMod(dsp.mod.SNEAK_DURATION)
```

Comment:
This should only increase the duration on the wearer. Also, it should be added into the base duration, not tacked on after a level check (won't matter because you can't cast it on yourself if you're level synced below the spell level)